### PR TITLE
Add analytics service

### PR DESCRIPTION
**Pull request by sophiebrown01**
### Context

This commits adds an `AnalyticsUpdateService`, which keeps our spreadsheet of who is using registers up-to-date.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -21,6 +21,8 @@ def create
       redirect_to @user, notice: 'User was successfully created.'
       NotifyMailer.api_key_confirmation(@user).deliver_now
       NotifyMailer.new_api_key_request(@user).deliver_now
+      analytics_update_service = AnalyticsUpdateService.new
```
Comment (gidsg) [inline app/controllers/users_controller.rb]: I wonder if we could move this to an ActiveJob? Then we can `perform_later` and run it async on the queue server.

Comment (sophiebrown01) [reply app/controllers/users_controller.rb]: As discussed.

File: app/services/analytics_update_service.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    authorization = Google::Auth::ServiceAccountCredentials.make_creds(
+        json_key_io: StringIO.new(Base64.decode64(ENV['GOOGLE_CREDENTIALS_JSON'])),
+        scope: [Google::Apis::SheetsV4::AUTH_SPREADSHEETS, Google::Apis::AnalyticsV3::AUTH_ANALYTICS]
+    )
+
+    @service = Google::Apis::SheetsV4::SheetsService.new
+    @service.authorization = authorization
+  end
+
+  def add_new_service(api_key, department = 'Not available', service = 'Not available', options = { spreadsheet_id: ENV['SPREADSHEET_ID'], range: ENV['SPREADSHEET_RANGE'] })
```
Comment (gidsg) [inline app/services/analytics_update_service.rb]: do we want to move getting the these `ENV` variables into Rails config?

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (dankmitchell) [reply app/controllers/users_controller.rb]: Will this work? `AnalyticsUpdateService.new({@user[:api_key] @user[:department] @user[:service]}).add_new_service `

Comment (sophiebrown01) [reply app/controllers/users_controller.rb]: I don't see any reason why not

File: app/services/analytics_update_service.rb (hunk 2 above)
Comment (dankmitchell) [reply app/services/analytics_update_service.rb]: Are they secret?

Comment (sophiebrown01) [reply app/services/analytics_update_service.rb]: Not particularly. The spreadsheet id will identify the spreadsheet we're using but it's behind access control anyway.

Comment (dankmitchell) [reply app/services/analytics_update_service.rb]: Then I agree with Gids 👍

**Review by dankmitchell — APPROVED** [review]

**Pull request merged**